clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -254,6 +297,13 @@ synchronized void registerMetric(KafkaMetric metric) {
      */
     @Override
     public void close() {
+        this.scheduler.shutdown();
+        try {
+            this.scheduler.awaitTermination(30, TimeUnit.SECONDS);
```

Comment:
We should ideally set these threads as daemon to avoid the possibility of hanging shutdowns in clients/server. This requires associating a thread factory with the executor as we do in the server’s `KafkaScheduler`. This will also get rid of the requirement to formally close the metrics object in all the tests (although I think close is a good practice regardless if we are associating a thread pool with metrics).